### PR TITLE
2066_notification

### PR DESCRIPTION
**Pull request by hoanglong96**
# SUMMARY
- Notification
# Redmine 
https://edu-redmine.framgia.vn/issues/2066
# BUGLIST

### CODE REVIEW

**Comment by HoangLongBFramgia** [comment]
ready

**Comment by HoangLongBFramgia** [comment]
ready

File: app/src/main/java/music_38/framgia/com/musicup/service/SongService.java [hunk 1]
```diff
@@ -144,4 +180,169 @@ public void setOnMediaPlayerChangeListener(SongServiceContract.OnMediaPlayerChan
...
+                mNotification,
+                ID_NOTIFICATION);
+        NotificationTarget notificationTargetBackGround = new NotificationTarget(
+                getApplicationContext(),
+                R.id.image_background_notification,
+                mRemoteViews,
+                mNotification,
+                ID_NOTIFICATION);
+        if (url != null && !url.equals("null")) {
+            Glide.with(getApplicationContext())
```
Comment (DoanVanToan) [inline app/src/main/java/music_38/framgia/com/musicup/service/SongService.java]: e có thể sử dụng 
Glide.with(getApplicationContext())
                    .asBitmap()
                    .load(url)
                    .error(R.drawable.image_default)
                    .apply(bitmapTransform(new BlurTransformation(25, 25)))
                    .into(notificationTargetBackGround);

ko cần check ntn

File: app/src/main/java/music_38/framgia/com/musicup/service/SongService.java [hunk 2]
```diff
@@ -144,4 +180,169 @@ public void setOnMediaPlayerChangeListener(SongServiceContract.OnMediaPlayerChan
...
+        }
+        startForeground(ID_NOTIFICATION, mNotification);
+    }
+
+    public class MusicIntentReceiver extends BroadcastReceiver {
+        private int mCount = SharedPrefs.getInstance().get(PREF_HEADSET_PLUG, Integer.class);
+
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            if (intent.getAction() == null) {
```
Comment (DoanVanToan) [inline app/src/main/java/music_38/framgia/com/musicup/service/SongService.java]: check if intent.getAction() == null || mSongManager == null return

**Comment by HoangLongBFramgia** [comment]
ready

**Pull request merged**